mongodbatlas/resource_mongodbatlas_advanced_cluster.go

```diff
@@ -865,13 +889,21 @@ func flattenAdvancedReplicationSpecs(apiObjects []*matlas.AdvancedReplicationSpe
 
 		if len(tfMapObjects) > 0 {
 			tfMapObject := tfMapObjects[i].(map[string]interface{})
-			tfList = append(tfList, flattenAdvancedReplicationSpec(apiObject, tfMapObject))
+			object, err := flattenAdvancedReplicationSpec(ctx, apiObject, tfMapObject, d, conn)
```

Comment:
`tfMapObject` can be nil right? `flattenAdvancedReplicationSpec` actually returns a nil, so can this be done in a single operation instead of doing this twice?